File: java/Sample.java
//import com.tuz358.matrix;

class Sample {
  public static void main(String args[]){
    float[][] a = {{1, 2, 3}, {4, 5, 6}, {7, 8, 9}};
    float[][] b = {{1, 0, 0}, {0, 1, 0}, {0, 0, 1}};

    matrix A = new matrix(a);
    matrix B = new matrix(b);
    matrix C = new matrix(3, 3);

    System.out.println("C = A.dot(B)");
    System.out.println("------------------");
    C = A.dot(B);
    printmatrix(A, "A");
    printmatrix(B, "B");
    printmatrix(C, "C");
    System.out.println("------------------");

    System.out.println("A_t = A.transpose()");
    System.out.println("------------------");
    matrix A_t = A.transpose();
    printmatrix(A_t, "A_t");
    System.out.println("------------------");

    System.out.println("C = A.outer(B)");
    System.out.println("------------------");
    C = A.outer(B);
    printmatrix(C, "C");
    System.out.println("------------------");

    System.out.println("tr_A = A.trace()");
    System.out.println("------------------");
    float tr_A = A.trace();
    System.out.println(java.text.MessageFormat.format("tr_A: {0}\n", tr_A));
    System.out.println("------------------");

    System.out.println("E3 = new matrix(3, 3).identity(3)");
    System.out.println("------------------");
    matrix E3 = new matrix(3, 3).identity(3);
    printmatrix(E3, "E3");
    System.out.println("------------------");

    System.out.println("C = A.multiply(B)");
    System.out.println("------------------");
    C = A.multiply(B);
    printmatrix(C, "C");
    System.out.println("------------------");

    System.out.println("C = A.direct_sum(B)");
    System.out.println("------------------");
    C = A.direct_sum(B);
    printmatrix(C, "C");
    System.out.println("------------------");
  }

  private static void printmatrix(matrix x, String name){
    System.out.println("matrix " + name + ":");
    for (int i = 0;i < x.data.length;i++) {
      for (int j = 0;j < x.data[0].length;j++) {
        System.out.print(String.valueOf(x.data[i][j]) + '\t');
      }
      System.out.println();
    }
    System.out.println();
  }

}

File: php/sample.php
<?php
require('matrix.php');

$a = array(array(1, 2, 3), array(4, 5, 6), array(7, 8, 9));
$b = array(array(1, 0, 0), array(0, 1, 0), array(0, 0, 1));

$matrix = new Matrix($a);


function printmatrix($data) {
  for ($i = 0; $i < count($data); $i++) {
    for ($j = 0; $j < count($data[0]); $j++) {
      echo $data[$i][$j].' ';
    }
    echo "\n";
  }
  echo "\n";
}

echo "matrix a:\n";
printmatrix($a);
echo "matrix b:\n";
printmatrix($b);

echo "dot(a, b):\n";
$out = $matrix->dot($b);
printmatrix($out);

echo "transpose(a):\n";
printmatrix($matrix->transpose());

echo "outer(a, b):\n";
printmatrix($matrix->outer($b));

echo "trace(a): ";
print($matrix->trace());
echo "\n";

echo "identity(3):\n";
printmatrix($matrix->identity(3));

echo "multiply(a, b):\n";
printmatrix($matrix->multiply($b));

echo "direct_sum(a, b):\n";
printmatrix($matrix->direct_sum($b));

 ?>
